src/main/java/nl/tudelft/ewi/devhub/server/web/resources/ProjectAssignmentsResource.java

```diff
@@ -0,0 +1,312 @@
+                                  @PathParam("assignmentId") Long assignmentId,
+                                  @PathParam("path") String attachmentPath) {
+
+        Assignment assignment = assignments.find(group.getCourse(), assignmentId);
+        File file = deliveriesBackend.getAttachment(assignment, group, attachmentPath);
+        return Response.ok(file, MediaType.APPLICATION_OCTET_STREAM).build();
+    }
+
+    /**
+     * Review a specific delivery
+     * @param request the current HttpServletRequest
+     * @param deliveryId the delivery id
+     * @return a view for reviewing the delivery
+     */
+    @GET
+    @Transactional
+    @Path("deliveries/{deliveryId}/review")
+    public Response getReviewView(@Context HttpServletRequest request,
+                                  @PathParam("deliveryId") Long deliveryId)
+            throws ApiError, IOException, GitClientException {
+
+        if(!(currentUser.isAdmin() || currentUser.isAssisting(group.getCourse()))) {
+            throw new UnauthorizedException();
+        }
+
+        Delivery delivery = deliveries.find(group, deliveryId);
+
+        Map<String, Object> parameters = Maps.newLinkedHashMap();
+        parameters.put("user", currentUser);
+        parameters.put("group", group);
+        parameters.put("course", group.getCourse());
+        parameters.put("delivery", delivery);
+        parameters.put("assignment", delivery.getAssignment());
+        parameters.put("deliveryStates", Delivery.State.values());
+        parameters.put("states", new ProjectResource.CommitChecker(group, buildResults));
+        parameters.put("repository", gitClient.repositories().retrieve(group.getRepositoryName()));
+
+        List<Locale> locales = Collections.list(request.getLocales());
+        return display(templateEngine.process("courses/assignments/group-delivery-review.ftl", locales, parameters));
+    }
+
+    /**
+     * Submit a review for a delivery
+     * @param request the current HttpServletRequest
+     * @param deliveryId the delivery id
+     * @return a redirect to the deliveries
+     */
+    @POST
+    @Path("deliveries/{deliveryId}/review")
+    public Response processReview(@Context HttpServletRequest request,
+                                  @PathParam("deliveryId") Long deliveryId,
+                                  @FormParam("grade") String grade,
+                                  @FormParam("commentary") String commentary,
+                                  @FormParam("state") Delivery.State state) throws UnauthorizedException, ApiError {
+
+        if(!(currentUser.isAdmin() || currentUser.isAssisting(group.getCourse()))) {
+            throw new UnauthorizedException();
+        }
+
+        Double gradeValue = grade.isEmpty() ? null : Double.valueOf(grade);
+        Delivery delivery = deliveries.find(group, deliveryId);
+        Delivery.Review review = new Delivery.Review();
+        review.setState(state);
+        review.setGrade(gradeValue);
```

Comment:
This would be a constraint not necessarily true. For instance, you could postpone grading, because you reject an assignment but request only a small change. I agree that it doesn't really make sense to approve an assignment with a `3`, or disapprove an assignment with a `10`. However, this is only the case when the grades are from 1 to 10, and approved is for >5,5. This is not always the case.